tests/stress_tests/scripts/run_memcheck.py

```diff
@@ -90,15 +97,15 @@ def main():
     parser.add_argument('--gtest_parallel', help='Path to gtest-parallel to use.',
                         default='gtest_parallel')
     parser.add_argument('-d', '--output_dir',
-                        required=args.timeline_report or args.upload,
+                        required=args.timeline_report or args.upload or args.compare,
```

Comment:
args.compare does not need DB if it is a folder.  Suggest
```
is_database_required = args.timeline_report or args.upload or (args.compare and os.path.isdir(args.compare))
...
required=is_database_required
...
required=is_database_required
...
required=is_database_required

```

Reply:
Good catch. But I decided to not combine such condition in one variable because it's hard to apply - every case is unique enough